bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/form/FormStructureParserImpl.java

```diff
@@ -43,6 +43,26 @@ public String getClientLibRefFromFormContainer() {
         return getPropertyFromFormContainer(resource, FormContainer.PN_CLIENT_LIB_REF);
     }
 
+    @Override
+    public Boolean containsFormContainer() {
+        return containsFormContainer(resource);
+    }
+
+    private Boolean containsFormContainer(Resource resource) {
```

Comment:
Why did you create another method, it is confusing. Can you add it the original method ?

Reply:
We'll need to recursively traverse the children nodes to check whether there exists a resource with fieldType is form. Its a private recursive method.